parsel/selector.py

```diff
@@ -276,23 +283,29 @@ def re_first(self, regex, default=None, replace_entities=True):
         """
         return next(iflatten(self.re(regex, replace_entities=replace_entities)), default)
 
-    def extract(self):
+    def extract(self, format_as=None):
         """
         Serialize and return the matched nodes in a single unicode string.
         Percent encoded content is unquoted.
+        
+        ``format_as`` can take either ``xml`` or ``html`` values to pretty format the output. 
         """
         try:
-            return etree.tostring(self.root,
-                                  method=self._tostring_method,
-                                  encoding='unicode',
-                                  with_tail=False)
+            return etree.tostring(
+                self.root,
+                method=format_as or self._tostring_method,
+                encoding='unicode',
+                with_tail=False,
+                pretty_print=bool(format_as),
```

Comment:
This doesn't look right, it's using the boolean value of `format_as`.
Can you please add tests for the full behavior? :)

Reply:
@eliasdorneles the logic behind this is that if `format_as` is explicitly specified then it should pretty_print as well. Alternatively pretty_print argument should be added separately I guess